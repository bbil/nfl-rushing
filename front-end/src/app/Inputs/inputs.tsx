import React from 'react';
import { connect } from 'react-redux';
import { getCsvLink } from '../../redux/rushing/selectors';
import { State } from '../../redux/store';
import { Filter } from './filter';
import { Sorting } from './sorting';

export interface InputsProps {
    csvLink: string;
}

export const InputsComponent: React.FC<InputsProps> = props => {
    return (
        <div className="container-fluid">
            <div className="row">
                <div className="col">
                    <Filter />
                </div>
                <div className="col">
                    <Sorting />
                </div>
                <div className="col">
                    <a href={props.csvLink}>Download CSV</a>
                </div>
            </div>
        </div>
    );
};

function mapStateToProps(state: State): InputsProps {
    return {
        csvLink: getCsvLink(state)
    }
}

export const Inputs = connect(mapStateToProps)(InputsComponent);