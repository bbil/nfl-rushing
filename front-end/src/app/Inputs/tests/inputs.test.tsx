import '@testing-library/react';

import { shallow } from 'enzyme';

import { InputsComponent, InputsProps } from '../inputs';

describe('Inputs component', () => {
    const PROPS: InputsProps = {
        csvLink: 'csvLink'
    };

    it('renders', () => {
        const wrapper = shallow(<InputsComponent {...PROPS} />);

        expect(wrapper).toMatchSnapshot();
    });
});
