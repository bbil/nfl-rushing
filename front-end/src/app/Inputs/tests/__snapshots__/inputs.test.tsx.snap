// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs component renders 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <Connect(FilterComponent) />
    </div>
    <div
      className="col"
    >
      <Connect(SortingComponent) />
    </div>
    <div
      className="col"
    >
      <a
        href="csvLink"
      >
        Download CSV
      </a>
    </div>
  </div>
</div>
`;
